src/pages/organisationUnits/list/OrganisationUnitList.tsx

```diff
@@ -125,26 +107,17 @@ export const OrganisationUnitList = () => {
 
     // expand ancestors of the filtered org units
     useEffect(() => {
+        // reset state when not filtering
         if (!isFiltering) {
-            setExpanded(inititalExpandedState)
-            return
-        }
-        if (!orgUnitFiltered.data) {
-            return
-        }
-        const ancestorIds = orgUnitFiltered.data?.organisationUnits.flatMap(
-            (ou) => ou.ancestors.map((a) => a.id)
-        )
-        if (!ancestorIds) {
+            setExpanded(initialExpandedState)
+            setParentIdsToLoad(initialExpandedState)
             return
         }
-        const expandedObj = Object.fromEntries(
-            ancestorIds.map((id) => [id, true])
-        )
-        setExpandedDuringFilter(expandedObj)
-        // this will "hide" data from useOrgUnitChildrenQueries, and only show the relevant data for the filter
-        setExpanded({})
-    }, [isFiltering, orgUnitFiltered.data, inititalExpandedState])
+        // if we are filtering, expand all, and reset parentIdsToLoad
+        setExpanded(true)
```

Comment:
We can now leverage built in "true" means show all, because `expanded` state is not also used directly.